common/src/app/common/pages/changes.cljc

```diff
@@ -385,15 +385,27 @@
 
 (defmethod process-change :mov-objects
   [data {:keys [parent-id shapes index page-id component-id ignore-touched after-shape]}]
-  (letfn [(calculate-invalid-targets [objects shape-id]
+  (letfn [(nested-components? [objects shape-id]
+             (let [children          (cph/get-children-with-self objects shape-id)
+                   child-components  (into #{} (keep :component-id children))
```

Comment:
`(into #{} (keep :component-id children))` can be replaced with: `(into #{} (keep :component-id) children)`

The `(keep :component-id)` can be also defined at namespace level for reuse like:

`(def xf-keep-component-id (keep :component-id))`

And then re-used:

`(into #{} xf-keep-component-id children)`

Reply:
done